src/modules/muorb/krait/px4muorb_KraitRpcWrapper.cpp

```diff
@@ -59,6 +63,73 @@ static const uint32_t _MAX_TOPICS = 64;
 static const uint32_t _MAX_BULK_TRANSFER_BUFFER_SIZE = _MAX_TOPIC_DATA_BUFFER_SIZE * _MAX_TOPICS;
 static uint8_t *_BulkTransferBuffer = 0;
 
+// The DSP timer can be read from this file.
+#define DSP_TIMER_FILE "/sys/kernel/boot_adsp/qdsp_qtimer"
+
+/**
+ * Helper function to get timer difference between time on DSP and appsproc side.
+ * Usually the DSP gets started around 2s before the appsproc (Linux) side and
+ * therefore the clocks are not in sync. We change the clock on the DSP side but
+ * for this we need to find the offset first and then tell code on the DSP side.
+ *
+ * @param time_diff_us: pointer to time offset to set.
+ * @return: 0 on success, < 0 on error.
+ */
+int calc_timer_diff_to_dsp_us(int32_t *time_diff_us);
+
+int calc_timer_diff_to_dsp_us(int32_t *time_diff_us)
+{
+	int fd = open(DSP_TIMER_FILE, O_RDONLY);
+
+	if (fd < 0) {
+		PX4_ERR("Could not open DSP timer file %s.", DSP_TIMER_FILE);
+		return -1;
+	}
+
+	char buffer[21];
+	memset(buffer, 0, sizeof(buffer));
+	int bytes_read = read(fd, buffer, sizeof(buffer));
+
+	if (bytes_read < 0) {
+		PX4_ERR("Could not read DSP timer file %s.", DSP_TIMER_FILE);
+		close(fd);
+		return -2;
+	}
+
+	// Do this call right after reading to avoid latency here.W
```

Comment:
Extra 'W'.

Might be worth repeating the measurement a few times and taking the median in case this code is unlucky enough to get scheduled out right here.